### PR TITLE
Add Java quickstart

### PR DESCRIPTION
**Pull request by ejona86**
As part of #237

### CODE REVIEW

File: docs/quickstarts/java-quickstart.md [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+## Download the example
+
+You'll need a local copy of the example code to work through this quickstart. Download the example code from our Github repository (the following command clones the entire repository, but you just need the examples for this quickstart and other tutorials):
+
+```sh
+  $ # Clone the repository to get the example code:
+  $ git clone https://github.com/grpc/grpc-java
+  $ # Navigate to the "hello, world" Python example:
+  $ cd grpc/examples/python/helloworld
```
Comment (hsaliak) [inline docs/quickstarts/java-quickstart.md]: Please fix.

File: docs/quickstarts/java-quickstart.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+<p class="lead">This guide gets you started with gRPC in Java with a simple working example.</p>
+
+<div id="toc"></div>
+
+## Before you begin
+
+### Prerequisites
+
+* `JDK`: version 7 or higher
+
```
Comment (hsaliak) [inline docs/quickstarts/java-quickstart.md]: Do we need to add gradle to the list here?

File: docs/quickstarts/java-quickstart.md [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+```
+
+(Don't forget to save the file!)
+
+## Update and run the application
+
+Normal compilation regenerates `GreeterGrpc.java`, which contains our generated client and server classes. It also regenerates classes for populating, serializing, and retrieving our request and response types.
+
+There's no need to worry about generating new server and client code, but we still need to implement and call the new method in the human-written parts of our example application.
+
```
Comment (LisaFC) [inline docs/quickstarts/java-quickstart.md]: These two paragraphs are a bit unclear - how about:

When you recompile the example, as you'll do later, normal compilation will regenerate `GreeterGrpc.java`, which contains our generated gRPC client and server classes. This also regenerates classes for populating, serializing, and retrieving our request and response types.

However, you still need to implement and call the new method in the human-written parts of our example application.

**Comment by LisaFC** [comment]
LGTM. Anyone else want to look at this before I merge it?

File: docs/quickstarts/java-quickstart.md [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+   ```
+
+Congratulations! You've just run a client-server application with gRPC.
+
+## Update a gRPC service
+
+Now let's look at how to update the application with an extra method on the
+server for the client to call. Our gRPC service is defined using protocol
+buffers; you can find out lots more about how to define a service in a `.proto`
+file in [What is gRPC?]() and [gRPC Basics: Java][]. For now all you need
```
Comment (hsaliak) [inline docs/quickstarts/java-quickstart.md]: What is the right link here for "What is gRPC"  @LisaFC @zinuga

**Comment by hsaliak** [comment]
LGTM - I have a question on where we are going to point the users to for the "What is gRPC" link, other than that, its good.

File: docs/quickstarts/java-quickstart.md (hunk 4 above)
Comment (hsaliak) [inline docs/quickstarts/java-quickstart.md]: Discussed with @zinuga: We can drop the reference to What is gRPC? here to simplify.

**Comment by ejona86** [comment]
@hsaliak PTAL

**Comment by hsaliak** [comment]
LGTM

**Comment by ejona86** [comment]
@LisaFC, looks like it is ready to merge.

**Pull request merged**